### DISCUSSION TITLE
issues when upserting several thousand records - is there a better way?

### DISCUSSION THREAD

**Discussion by samthomson** [prisma/prisma#5319] [Q&A]
Hi, thanks for the great library :)

I have around 3500 records to insert into a - mysql - DB. I want to check they aren't there already, and only insert the 'new' ones. For this I am using prisma's `upsert` method:
```ts
    await Promise.all(
        currencies.map(async ({ name, symbol }) => {
            try {
                // await prisma.$connect()
                await prisma.currency.upsert({
                    where: { symbol },
                    update: { name, symbol },
                    create: { name, symbol },
                })
                // await prisma.$disconnect()
            } catch (err) {
                console.error('err adding', err)
            }
        }),
    )
    await prisma.$disconnect()
```
I thought by blocking via async/await would ensure I wasn't straining the database unduly. However I get a stream of these errors:
![image](https://user-images.githubusercontent.com/6588340/106121577-9f6d8e80-614f-11eb-8bee-334f2afef82c.png)

As you can see in my code, I have tried connecting/disconnecting explicitly to ensure a connection from one upsert doesn't hang around and saturate subsequent upserts use of a/any connection. It didn't change anything.

I have then tried the code again, substituting prisma for sequelize and using the `[model].findOrCreate` method instead of prismas `upsert`. It works perfectly, so I am wondering how in Prisma do I configure my connection [pool?] to allow for so many sequential operations OR is there a way to bulk/batch queries? 
Ideally I'd know how to run the above code without error (just so I can be confident in this ORM) but then also learn how to batch them, just for efficiency of the program.

**Comment by ryands17** [comment]
@samthomson 👋 
I would suggest creating an issue for this [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=). As a workaround, could you try running `Promise.all` for records in batches of 500 or 1000?

**Reply by samthomson** [reply]
Hi @ryands17 thanks, I'll do just that. done https://github.com/prisma/prisma/issues/5320

**Reply by samthomson** [reply]
for now I will just use sequelize for this operation as the workaround, and when there is a solution I'll implement that and keep on with prisma.

**Reply by janpio** [reply]
This should be interesting: https://github.com/prisma/prisma/issues/4998#issuecomment-766742955

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
